src/syrupy/assertion.py

```diff
@@ -90,23 +91,22 @@ def __eq__(self, other) -> bool:
     def _assert(self, data) -> bool:
         self._session.register_assertion(self)
         try:
-            if self._update_snapshots:
-                serialized_data = self.serializer.encode(data)
+            serialized_data = self.serializer.encode(data)
+            snapshot_data = self._recall_data(index=self.num_executions)
+            matches = snapshot_data is not None and serialized_data == snapshot_data
+            if matches:
+                return True
+            if not matches and self._update_snapshots:
```

Comment:
(nb) if you move this condition before line 97, you can then just return `matches`. I.e. 2 if conditions instead of 3.

Reply:
Will do this